pkg/etcd/etcd.go

```diff
@@ -495,8 +496,16 @@ func (e *ETCD) removePeer(ctx context.Context, id, address string) error {
 				return err
 			}
 			if u.Hostname() == address {
+				if e.address == address {
+					continue
+				}
 				logrus.Infof("Removing name=%s id=%d address=%s from etcd", member.Name, member.ID, address)
 				_, err := e.client.MemberRemove(ctx, member.ID)
+				if err != nil {
+					if err == rpctypes.ErrGRPCMemberNotFound {
+						return nil
+					}
+				}
```

Comment:
since we return err anyway down below, can probably just reduce this to:

```suggestion
				if err == rpctypes.ErrGRPCMemberNotFound {
					return nil
				}
```